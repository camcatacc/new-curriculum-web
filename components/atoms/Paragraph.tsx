import { Typography } from "@material-ui/core";
import React from "react";

export const Paragraph = (props: { children: any }) => {
    return (
        <Typography variant="body1" align="justify" style={{ lineHeight: 2.2 }}>
            {props.children}
        </Typography>
    );
};
