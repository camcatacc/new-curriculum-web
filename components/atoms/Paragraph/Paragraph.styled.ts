// Modules
import styled from "styled-components";

// Elements
import { Typography as MUTypography } from "@material-ui/core";

// Components
export const Typography = styled(MUTypography)`
	line-height: 2;
`;
