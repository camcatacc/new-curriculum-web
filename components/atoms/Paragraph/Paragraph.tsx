// Modules
import React from "react";

// Elements
import { Typography } from "@material-ui/core";

// Definitions
export interface ParagraphProps {
	children: React.ReactNode;
}

// Element
const Paragraph = ({ children }: ParagraphProps) => {
	return (
		<Typography variant="body1" align="justify" style={{ lineHeight: 2 }}>
			{children}
		</Typography>
	);
};

export default Paragraph;
