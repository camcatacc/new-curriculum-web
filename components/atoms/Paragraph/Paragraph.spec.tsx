// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import Paragraph from "components/atoms/Paragraph/Paragraph";

// Definitions
import type { ParagraphProps } from "components/atoms/Paragraph/Paragraph";

// Default props
const defaultProps: ParagraphProps = {
	children: <p>Bu</p>
};

// Tests
describe("Paragraph", () => {
	it("Renders as expected", () => {
		const { container } = render(<Paragraph {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
