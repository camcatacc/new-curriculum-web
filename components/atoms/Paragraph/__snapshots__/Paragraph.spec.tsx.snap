// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Renders as expected 1`] = `
<div>
  <p
    class="MuiTypography-root sc-bdvvtL gedMyf MuiTypography-body1 MuiTypography-alignJustify"
  >
    <p>
      Bu
    </p>
  </p>
</div>
`;
